OpenSim/Actuators/VandenBogert2011Muscle.h

```diff
@@ -0,0 +1,568 @@
+
+        /** @param the activation time constant, T_{act} (s)*/
+        void setActivTimeConstant(double activTimeConstant);
+        /** @returns the activation time constant, T_{act} (s)*/
+        double getActivTimeConstant() const;
+
+        /** @param the deactivation time constant, T_{deact} (s)*/
+        void setDeactivTimeConstant(double deactivTimeConstant);
+        /** @returns the deactivation time constant, T_{deact} (s)*/
+        double getDeactivTimeConstant() const;
+
+        /** @param the pennation angel at optimal fiber length,
+        phi_{opt} (rad)*/
+        void setPennAtOptFiberLength(double pennAtOptFiberLength);
+        /** @returns the pennation angel at optimal fiber length,
+        phi_{opt} (rad)*/
+        double getPennAtOptFiberLength() const;
+
+        /** @param the default muscle activation, (dimensionless)*/
+        void setDefaultActivation(double defaultActivation);
+        /** @returns the default muscle activation, (dimensionless)*/
+        double getDefaultActivation() const;
+
+        /** @param the default muscle fiber length, ()*/
+        void setDefaultFiberLength(double fiberLength);
+        /** @returns the default muscle fiber length,, ()*/
+        double getDefaultFiberLength() const;
+
+
+        /**@param s The state of the system.
+        @param the muscle activation, (dimensionless)*/
+        void setActivation(SimTK::State& s,double activation) const override;
+        /**@param s The state of the system.
+        @param fiberLength the muscle fiber length, ()*/
+        void setFiberLength(SimTK::State& s,double fiberLength) const;
+
+        /* Calculate the muscle implicit residual.  Returns a state Vec2
+        containing the force residual and activation residual.  When these
+        value are equal to zero, the muscle equations are "balanced" and the
+         muscle  is muscle is in a valid state.
+                @param s The state of the system.
+                @param projFibVelNorm_guess The muscle fiber velocity, projected
+                in line with tendon and normalized by the optimal fiber length,
+                (dimensionless)
+                @param activdot_guess The muscle activation time derivative,
+                (1/s)
+                @returns a vector of the muscle residual:
+                        \li 0: force (N)
+                        \li 1: activation (unitless)*/
+        SimTK::Vec2 getResidual(const SimTK::State& s,
+                                double projFibVelNorm_guess,
+                                double activdot_guess, double u) const;
+
+
+        //Hacks because cache variables not implemented yet
+
+        /**@param The state of the system.
+           @returns The Fiber Length of the muscle.*/
+        double  getFiberLength(SimTK::State& s) const;
+        /**@param The state of the system.
+           @returns The activation of the muscle.*/
+        double  getActivation(SimTK::State& s) const;
```

Comment:
This is overriding a virtual method, correct? I think you could use the `override` keyword here to make sure you are overriding the method.